internal/backends/sqlite/database.go

```diff
@@ -102,8 +102,75 @@ func (db *database) RenameCollection(ctx context.Context, params *backends.Renam
 }
 
 // Stats implements backends.Database interface.
+//
+// If the database does not exist, it returns *backends.DBStatsResult filled with zeros for all the fields.
 func (db *database) Stats(ctx context.Context, params *backends.StatsParams) (*backends.StatsResult, error) {
-	panic("not implemented")
+	stats := new(backends.StatsResult)
+
+	d := db.r.DatabaseGetExisting(ctx, db.name)
+	if d == nil {
+		return stats, nil
+	}
+
+	// Call ANALYZE to update statistics of tables and indexes,
+	// see https://www.sqlite.org/lang_analyze.html.
+	q := `ANALYZE`
+	if _, err := d.ExecContext(ctx, q); err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	// Total size is the disk space used by the database,
+	// see https://www.sqlite.org/dbstat.html.
+	q = `
+		SELECT
+			SUM(pgsize)
+		FROM dbstat WHERE aggregate = TRUE`
+
+	err := d.QueryRowContext(ctx, q).Scan(&stats.SizeTotal)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	// Count, size and cells of tables exclude sqlite internal tables and FerretDB meta table.
+	// It uses number of cells to approximate total row count.
+	// See https://www.sqlite.org/schematab.html and
+	// also https://www.sqlite.org/fileformat.html.
+	q = `
+		SELECT
+		    COUNT(s.name)             AS CountTables,
+		    COALESCE(SUM(d.pgsize),0) AS SizeTables,
+		    COALESCE(SUM(d.ncell),0)  AS CountCells
+		FROM sqlite_schema AS s
+			LEFT JOIN dbstat AS d ON d.name = s.tbl_name
+		WHERE s.type = 'table' AND s.name NOT LIKE ? AND s.name <> ?`
```

Comment:
Another option is to get a list of collections/tables (see #3292) and use `d.name IN (?, ?, ?, ...)`. After all, the registry is responsible for maintaining of collections/tables, not `sqlite_schema`

Reply:
That is a really good point. There's another advantage of doing that way, I think I can use sqlite aggregation query if I avoid joining with sqlite_schema table. Plus registry should give me the number of tables without querying!